src/jdk.crypto.ec/share/classes/sun/security/ec/ECDSAOperations.java

```diff
@@ -256,10 +257,8 @@ public boolean verifySignedDigest(byte[] digest, byte[] sig, ECPoint pp) {
 
         ecOps.setSum(p1, p2.asAffine());
         IntegerModuloP result = p1.asAffine().getX();
-        result = result.additiveInverse().add(ri);
-
         b2a(result, orderField, temp1);
-        return ECOperations.allZero(temp1);
+        return MessageDigest.isEqual(temp1, r);
```

Comment:
I did not get the point of this update.  Is it the broken case you mentioned in the PR description?  What's the issue of the original code?